tests/pytests/test_tags.py

```diff
@@ -374,3 +374,403 @@ def test_empty_suffix_withsuffixtrie(env):
+    env.assertEqual(res, expected_result)
+
+    res = env.cmd('FT.SEARCH', 'idx', "@tag:{trailing:space *}", 'DIALECT', 5)
+    env.assertEqual(res, expected_result)
+
+def testInvalidSyntax(env):
+    # Create index
+    env.expect('FT.CREATE', 'idx', 'ON', 'HASH', 'PREFIX', '1', '{doc}:',
+               'SCHEMA', 'tag', 'TAG', 'SORTABLE', 'id',
+               'NUMERIC', 'SORTABLE').ok()
+    waitForIndex(env, 'idx')
+
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd('FT.EXPLAIN', 'idx', "@tag:{ w'?*1'}", 'DIALECT', 5)
+
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd('FT.EXPLAIN', 'idx', "@tag:{w'?*1' }", 'DIALECT', 5)
+
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd('FT.EXPLAIN', 'idx', "@tag:{w\\'?*1'}", 'DIALECT', 5)
+
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd('FT.SEARCH', 'idx', '@tag:{abc:1\\}', 'DIALECT', 5)
+
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd('FT.SEARCH', 'idx', "@tag:{w'1?'*}", 'DIALECT', 5)
+
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd('FT.SEARCH', 'idx', "@tag:{*w'1?'}", 'DIALECT', 5)
+
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd('FT.SEARCH', 'idx', "@tag:{*w'1?'*}", 'DIALECT', 5)
+
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd('FT.SEARCH', 'idx', "@tag:{*\\w'abc'\\*}", 'DIALECT', 5)
+    
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd("FT.SEARCH idx @tag:{w'-abc*} DIALECT 5")
+
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd('FT.SEARCH', 'idx', "(@tag:{\\w'-abc*})", 'DIALECT', 5)
+         
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd("FT.SEARCH idx '@tag:{*w'-abc*}' DIALECT 5")
+
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd("FT.SEARCH idx @t1:(%) DIALECT 5")
+
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd("FT.SEARCH idx @t1:(|) DIALECT 5")
+
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd("FT.SEARCH idx @t1:({) DIALECT 5")
+
+    # test punct character
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd("FT.explaincli idx @t1:> DIALECT 5")
+
+    # test cntrl character
+    with env.assertResponseError(contained='Syntax error'):
+        env.cmd("FT.explaincli idx @t1:\10 DIALECT 5")
+
+def testTagUNF(env):
+    # Create sample data
+    env.cmd('HSET', '{doc}:1', 'tag', 'america')
+    env.cmd('HSET', '{doc}:2', 'tag', 'aMerica')
+    env.cmd('HSET', '{doc}:3', 'tag', 'America')
+
+    # Create index without UNF
+    env.expect('FT.CREATE', 'idx', 'ON', 'HASH', 'PREFIX', '1', '{doc}:',
+               'SCHEMA', 'tag', 'TAG', 'SORTABLE').ok()
+    waitForIndex(env, 'idx')
+
+    # TODO: This fails for dialect >= 4
```

Comment:
How does it fail?

Reply:
I was expecting to get the results ordered by the key: `[3, '{doc}:1', '{doc}:2', '{doc}:3']`.
But it returns `[3, 'doc:2', 'doc:3', 'doc:1']`.
But maybe I'm wrong with the expected order, because I checked and it returns the same order using DIALECT 2, 4 and 5.

Reply:
It was fixed creating the index before the docs.

Reply:
Explanation: since indexing in the background could ingest the docs in various ordering, resulting with different doc ids, resulting with different ordering in the response.